### PR TITLE
SUEWS module: save user input and output for diagnostics

### LINKED ISSUE #212: SUEWS module: save user input and output for diagnostics
often users might not manage to report issues with exact input for developers to reproduce the issues.
so, we need to properly save the user input and output for diagnostics.

### PR DESCRIPTION
**Pull request by sunt05**
Fixes #212

### CODE REVIEW

**Comment by sunt05** [comment]
@biglimp please review my additions for this functionality in the SUEWS module. I tested this here and it's working well but some tweaks might be necessary to blend some *QGIS flavour* (e.g., message box, etc.).

File: suewsmodel/suews_wrapper.py [hunk 1]
```diff
@@ -50,6 +50,29 @@ def wrapper(pathtoplugin, year=None):
...
+            p_dir_dump.mkdir(parents=True)
+
+        str_now = pd.Timestamp.now().isoformat().split('.')[0]
+        ver_supy=sp.__version__
+        fn_dump = p_dir_dump/f"supy{ver_supy}_diag_{str_now}.h5"
+        df_forcing.to_hdf(fn_dump, key="df_forcing")
+        df_output.to_hdf(fn_dump, key="df_output")
+        df_state_final.to_hdf(fn_dump, key="df_state_final")
+        # tell user the location where `fn_dump` is saved
+        print(f"{fn_dump.resolve()} has been successfully created!")
```
Comment (sunt05) [inline suewsmodel/suews_wrapper.py:68]: this line may be removed: apparently a leftover when I was testing it under a pure python environment.

**Comment by biglimp** [comment]
Nice. As user messages regards to the SUEWS tools (SUEWSSimple and SUEWSAdvanced) happens outside of the suewswrapper.py (see e.g. SuewsSimple/suews_simple.py; line 593), I would prefer to put your suggested code outside of the wrapper. This should also added in the processing branch.

**Comment by sunt05** [comment]
> Nice. As user messages regards to the SUEWS tools (SUEWSSimple and SUEWSAdvanced) happens outside of the suewswrapper.py (see e.g. SuewsSimple/suews_simple.py; line 593), I would prefer to put your suggested code outside of the wrapper. This should also added in the processing branch.

yes, that sounds more sensible. As I'm not very familiar with that part, please go ahead with modifications: just keep working on the branch [`sunt05/issue212`](https://github.com/UMEP-dev/UMEP/tree/sunt05/issue212).

**Review by biglimp — APPROVED** [review]

**Pull request merged**